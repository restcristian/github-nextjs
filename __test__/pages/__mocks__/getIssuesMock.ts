import { GET_ISSUES_QUERY } from "@/graphql/queries";

export const getIssuesMock = [
    {
      request: {
        query: GET_ISSUES_QUERY,
        variables: {
          owner: "reactjs",
          name: "reactjs.org",
        },
      },
      result: {
        data: {
          repository: {
            issues: {
              edges: [
                {
                  node: {
                    title: "[Beta] Remove runtime CSS-in-JS",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "[Beta] DocSearch Title Gets Escaped",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "[Beta] Issue in the navigation of the React Docs Page",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "Redux and functional components missing in react js",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "The issue is i have a component which is having search and select ",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Clock examples use impure `.toLocaleTimeString()`",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Always exited to hear newer updates ðŸ”¥ React ðŸš€",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Switch CDN provider in documentation",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "dependencies",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Add popular open source community: PrimeReact ",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Update react testing doc section for react 18",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "React+Vite",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "Strict mode making newbie like me get confused on debugging.",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "How to avoid re-renders on history.push?",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title: "Update `AUTHORS` file",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "[Beta] Bug: Gutter text overlaps with sandpack code on mobile",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "Beta reactjs 404 & 500 Custom Error pages similarities",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      'Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page',
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "pass the set state function to a non-descendent component ",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
                {
                  node: {
                    title:
                      "Content issue on `reactjs.org/tutorial/tutorial.html`",
                    __typename: "Issue",
                  },
                  __typename: "IssueEdge",
                },
              ],
              __typename: "IssueConnection",
            },
            __typename: "Repository",
          },
        },
      },
    },
  ];