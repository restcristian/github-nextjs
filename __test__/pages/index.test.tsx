import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import Home from "@/pages";
import { getIssuesMock } from "./__mocks__/getIssuesMock";
describe("Home", () => {
  
  let component: any  = null;
  beforeEach(() => {
   component = render(
        <MockedProvider mocks={getIssuesMock} addTypename={false}>
          <Home />
        </MockedProvider>
      );
  })
  it("should render a title", async () => {
    expect(await screen.findByTestId("loading")).toBeInTheDocument();
    expect(await screen.findByTestId("title")).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  it('should render an issue-list', async () => {
    expect(await screen.findByTestId("loading")).toBeInTheDocument();
    expect(await screen.findByTestId("issue-list")).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  })
});
