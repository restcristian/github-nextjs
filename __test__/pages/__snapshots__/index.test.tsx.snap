// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a title 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h2
          data-testid="title"
        >
          ReactJS Issues
        </h2>
        <ul
          data-testid="issue-list"
        >
          <li>
            [Beta] Remove runtime CSS-in-JS
          </li>
          <li>
            [Beta] DocSearch Title Gets Escaped
          </li>
          <li>
            [Beta] Issue in the navigation of the React Docs Page
          </li>
          <li>
            Redux and functional components missing in react js
          </li>
          <li>
            The issue is i have a component which is having search and select 
          </li>
          <li>
            Clock examples use impure \`.toLocaleTimeString()\`
          </li>
          <li>
            Always exited to hear newer updates ðŸ”¥ React ðŸš€
          </li>
          <li>
            Switch CDN provider in documentation
          </li>
          <li>
            dependencies
          </li>
          <li>
            Add popular open source community: PrimeReact 
          </li>
          <li>
            Update react testing doc section for react 18
          </li>
          <li>
            React+Vite
          </li>
          <li>
            Strict mode making newbie like me get confused on debugging.
          </li>
          <li>
            How to avoid re-renders on history.push?
          </li>
          <li>
            Update \`AUTHORS\` file
          </li>
          <li>
            [Beta] Bug: Gutter text overlaps with sandpack code on mobile
          </li>
          <li>
            Beta reactjs 404 & 500 Custom Error pages similarities
          </li>
          <li>
            Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page
          </li>
          <li>
            pass the set state function to a non-descendent component 
          </li>
          <li>
            Content issue on \`reactjs.org/tutorial/tutorial.html\`
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h2
        data-testid="title"
      >
        ReactJS Issues
      </h2>
      <ul
        data-testid="issue-list"
      >
        <li>
          [Beta] Remove runtime CSS-in-JS
        </li>
        <li>
          [Beta] DocSearch Title Gets Escaped
        </li>
        <li>
          [Beta] Issue in the navigation of the React Docs Page
        </li>
        <li>
          Redux and functional components missing in react js
        </li>
        <li>
          The issue is i have a component which is having search and select 
        </li>
        <li>
          Clock examples use impure \`.toLocaleTimeString()\`
        </li>
        <li>
          Always exited to hear newer updates ðŸ”¥ React ðŸš€
        </li>
        <li>
          Switch CDN provider in documentation
        </li>
        <li>
          dependencies
        </li>
        <li>
          Add popular open source community: PrimeReact 
        </li>
        <li>
          Update react testing doc section for react 18
        </li>
        <li>
          React+Vite
        </li>
        <li>
          Strict mode making newbie like me get confused on debugging.
        </li>
        <li>
          How to avoid re-renders on history.push?
        </li>
        <li>
          Update \`AUTHORS\` file
        </li>
        <li>
          [Beta] Bug: Gutter text overlaps with sandpack code on mobile
        </li>
        <li>
          Beta reactjs 404 & 500 Custom Error pages similarities
        </li>
        <li>
          Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page
        </li>
        <li>
          pass the set state function to a non-descendent component 
        </li>
        <li>
          Content issue on \`reactjs.org/tutorial/tutorial.html\`
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home should render an issue-list 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h2
          data-testid="title"
        >
          ReactJS Issues
        </h2>
        <ul
          data-testid="issue-list"
        >
          <li>
            [Beta] Remove runtime CSS-in-JS
          </li>
          <li>
            [Beta] DocSearch Title Gets Escaped
          </li>
          <li>
            [Beta] Issue in the navigation of the React Docs Page
          </li>
          <li>
            Redux and functional components missing in react js
          </li>
          <li>
            The issue is i have a component which is having search and select 
          </li>
          <li>
            Clock examples use impure \`.toLocaleTimeString()\`
          </li>
          <li>
            Always exited to hear newer updates ðŸ”¥ React ðŸš€
          </li>
          <li>
            Switch CDN provider in documentation
          </li>
          <li>
            dependencies
          </li>
          <li>
            Add popular open source community: PrimeReact 
          </li>
          <li>
            Update react testing doc section for react 18
          </li>
          <li>
            React+Vite
          </li>
          <li>
            Strict mode making newbie like me get confused on debugging.
          </li>
          <li>
            How to avoid re-renders on history.push?
          </li>
          <li>
            Update \`AUTHORS\` file
          </li>
          <li>
            [Beta] Bug: Gutter text overlaps with sandpack code on mobile
          </li>
          <li>
            Beta reactjs 404 & 500 Custom Error pages similarities
          </li>
          <li>
            Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page
          </li>
          <li>
            pass the set state function to a non-descendent component 
          </li>
          <li>
            Content issue on \`reactjs.org/tutorial/tutorial.html\`
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h2
        data-testid="title"
      >
        ReactJS Issues
      </h2>
      <ul
        data-testid="issue-list"
      >
        <li>
          [Beta] Remove runtime CSS-in-JS
        </li>
        <li>
          [Beta] DocSearch Title Gets Escaped
        </li>
        <li>
          [Beta] Issue in the navigation of the React Docs Page
        </li>
        <li>
          Redux and functional components missing in react js
        </li>
        <li>
          The issue is i have a component which is having search and select 
        </li>
        <li>
          Clock examples use impure \`.toLocaleTimeString()\`
        </li>
        <li>
          Always exited to hear newer updates ðŸ”¥ React ðŸš€
        </li>
        <li>
          Switch CDN provider in documentation
        </li>
        <li>
          dependencies
        </li>
        <li>
          Add popular open source community: PrimeReact 
        </li>
        <li>
          Update react testing doc section for react 18
        </li>
        <li>
          React+Vite
        </li>
        <li>
          Strict mode making newbie like me get confused on debugging.
        </li>
        <li>
          How to avoid re-renders on history.push?
        </li>
        <li>
          Update \`AUTHORS\` file
        </li>
        <li>
          [Beta] Bug: Gutter text overlaps with sandpack code on mobile
        </li>
        <li>
          Beta reactjs 404 & 500 Custom Error pages similarities
        </li>
        <li>
          Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page
        </li>
        <li>
          pass the set state function to a non-descendent component 
        </li>
        <li>
          Content issue on \`reactjs.org/tutorial/tutorial.html\`
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
